modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -111,7 +120,9 @@ export default class TileLayer extends CompositeLayer {
     const tilesetChanged = this.state.frameNumber !== frameNumber;
 
     if (isLoaded && onViewportLoad && (loadingStateChanged || tilesetChanged)) {
-      onViewportLoad(tileset.selectedTiles.map(tile => tile.data));
+      onViewportLoad(tileset.selectedTiles.map(tile => tile.data), {
```

Comment:
Don't change this callback if it's unnecessary

Reply:
Ok, maybe I didn't understand your last comment in the RFC. The conclusion is:
- `onViewportLoad` is not going to change.
- The new `onViewportChange` call back is the only callback that is going to return the `getRenderedFeatures` function.

If that is correct there are two situations in which onViewportChange should be called:
- When the viewport changes and all tiles have been loaded, for example when we drag the map a little and it doesn't loads new tiles. We can control it in `updateState` function checking `if (changeFlags.viewportChanged && tileset.isLoaded)`
- Every time a new set of tiles have finished loading for the current viewport, for example, when zoom ends `changeFlags.viewportChanged` is true but tiles are still loading so `onViewportChange` should be called when tiles have finished loading. It's being controlled in `_updateTileset` with the condition `if (isLoaded && (loadingStateChanged || tilesetChanged))`

In conclusion, in the new commit I'm calling the `onViewportChange` callback in these two situations.

Sorry for the long comment.

Reply:
Should it be called when the view changes (without triggering tile reloads)?

Reply:
> there are two situations in which onViewportChange should be called

That sounds right to me. If the viewport changes but not all tiles have been loaded, the callback is expected to trigger, and it's just delayed.

> Should it be called when the view changes?

I'd say yes, given that it affects the result of `getRenderedFeatures`

Reply:
Yes because if for example, you drag the map a little or your current zoom is 8 and you change it to zoom 8.4 it will not trigger a tile reloads but the features that you are seeing in the new viewport have probably changed. 

I'm trying to upload a video to show this but Github is returning an error. You can download it from [here](https://drive.google.com/file/d/1Vfts4PJyWqb0RfMDaW4weXzkveGvDaHu/view?usp=sharing) 

As you can see, in the browser tab new tiles are not downloaded and this condition `isLoaded && (loadingStateChanged || tilesetChanged)` is not satisfied.